be/src/runtime/data_stream_recvr.cc

```diff
@@ -471,12 +473,21 @@ Status DataStreamRecvr::SenderQueue::add_chunks(const PTransmitChunkParams& requ
     ChunkQueue chunks;
     size_t total_chunk_bytes = 0;
     faststring uncompressed_buffer;
-    bool prev_is_pipeline_level_shuffle = _is_pipeline_level_shuffle;
+
+#ifdef DEBUG
+    {
+        std::lock_guard<std::mutex> l(_lock);
```

Comment:
Do we really need to keep so many codes only for debugging?

Reply:
These code is only for DCHECK, the logic of pipeline level shuffle is too implied and complex, and I think adding `DCHECK` can expose problems as soon as possible. Otherwise, once a change is introduced, it will be difficult to troubleshoot if there exist bugs